Yona/Yona/Friends/AddFriendsViewController.swift

```diff
@@ -172,30 +172,44 @@ class AddFriendsViewController: UIViewController, UIScrollViewDelegate, UINaviga
             
         } else {
             
-            let postBuddyBody: [String:AnyObject] = [
-                postBuddyBodyKeys.sendingStatus.rawValue: buddyRequestStatus.REQUESTED.rawValue,
-                postBuddyBodyKeys.receivingStatus.rawValue: buddyRequestStatus.REQUESTED.rawValue,
-                postBuddyBodyKeys.message.rawValue : "Hi there, would you want to become my buddy?",
-                postBuddyBodyKeys.embedded.rawValue: [
-                    postBuddyBodyKeys.yonaUser.rawValue: [  //this is the details of the person you are adding
-                        addUserKeys.emailAddress.rawValue: emailTextfield.text ?? "",
-                        addUserKeys.firstNameKey.rawValue: firstnameTextfield.text ?? "",
-                        addUserKeys.lastNameKeys.rawValue: lastnameTextfield.text ?? "",
-                        addUserKeys.mobileNumberKeys.rawValue: mobileTextfield.text ?? ""  //this is the number of the person you are adding as a buddy
+            var number = ""
+            if let mobilenum = mobileTextfield.text {
+                number = (nederlandPhonePrefix) + mobilenum
+                
+                let trimmedWhiteSpaceString = number.removeWhitespace()
+                let trimmedString = trimmedWhiteSpaceString.removeBrackets()
+                
+                if trimmedString.validateMobileNumber() == false {
+                    self.displayAlertMessage("", alertDescription:
+                        NSLocalizedString("enter-number-validation", comment: ""))
+                    
+                }
+                let postBuddyBody: [String:AnyObject] = [
+                    postBuddyBodyKeys.sendingStatus.rawValue: buddyRequestStatus.REQUESTED.rawValue,
+                    postBuddyBodyKeys.receivingStatus.rawValue: buddyRequestStatus.REQUESTED.rawValue,
+                    postBuddyBodyKeys.message.rawValue : "Hi there, would you want to become my buddy?",
```

Comment:
This string is not localized yet!
`postBuddyBodyKeys.message.rawValue : "Hi there, would you want to become my buddy?",`